bahmni-mart/dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM openjdk:8-jre-alpine
+RUN apk add postgresql-client gettext tzdata && \
+    addgroup -S bahmni && \
+    adduser -S bahmni -G bahmni && \
+    mkdir -p /opt/bahmni-mart/log/ /opt/bahmni-mart/properties/ /var/www/bahmni_config/ /opt/bahmni-mart/conf/ /data/analytics /opt/bahmni-mart/lib/ && \
+    ln -s /opt/bahmni-mart/bin/bahmni-mart.sh /usr/bin/bahmni-mart && \
+    ln -s /opt/bahmni-mart/log /var/log/bahmni-mart && \
+    chown -R bahmni:bahmni /usr/bin/bahmni-mart /opt/bahmni-mart /var/log/bahmni-mart
+
+ARG TIMEZONE
+ADD scripts/ /etc/scripts/
+# ADD files/ /etc/files/
+COPY files/logback-spring.xml /opt/bahmni-mart/properties/logback-spring.xml
+COPY files/application.properties /tmp/application.properties
+COPY files/liquibase.xml files/bahmni-mart.json /opt/bahmni-mart/conf/
+RUN chown bahmni:bahmni /opt/bahmni-mart/properties/logback-spring.xml  /opt/bahmni-mart/conf/liquibase.xml && \
+    chmod 644 /opt/bahmni-mart/properties/logback-spring.xml && \
+    chmod 744 /opt/bahmni-mart/conf/liquibase.xml
+COPY files/bahmni-mart-1.0.0.jar /opt/bahmni-mart/lib/bahmni-mart.jar
+RUN chmod +x /etc/scripts/* && \
+    cp /usr/share/zoneinfo/$TIMEZONE /etc/localtime && \
+    echo "$TIMEZONE" > /etc/timezone && \
+apk del tzdata
+CMD ["/bin/sh", "/etc/scripts/setup.sh"]
```

Comment:
I think the good practice is to set a `USER` with a hardcoded UID.
Though only if the application does not require root access. I believe that's not the case here right?

Reply:
@rbuisson - Sorry I was not aware of this.Could you please let me know if we use autogenerated UID/GID - would we have any security concern?

Sorry for my ignorance :)

Reply:
Haha, I recently discovered that as well. Overall I find this guide a very good reference:
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user

I presume the bahmni-mart JAR does not need **root** access right?

Reply:
Hi, I add UID for the user in the commit below.

If I change USER to bahmni - I am facing problem while adding the cronjob.  I tried some work arounds, but could not find a proper solution. May need your help here @rbuisson

Reply:
As per our discussion, let's forget about running the `cron` as **bahmni**. I mean, ideally, it should, but that is acceptable if it does not.
However, I am surprised that we must still create a **bahmni** user and set the rights and owners.